fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushCountIntoUnionAll.java

```diff
@@ -0,0 +1,188 @@
+ *  +--LogicalUnion (outputs=[c1#13, count(c1)#19], regularChildrenOutputs=[[c1#9, count(c1)#16],
+ *   [a#4, count(a)#17], [a#7, count(a)#18]])
+ *    |--LogicalAggregate (groupByExpr=[c1#9], outputExpr=[c1#9, count(c1#9) AS `count(c1)`#16])
+ *    |  +--child1
+ *    |--LogicalAggregate (groupByExpr=[a#4], outputExpr=[a#4, count(a#4) AS `count(a)`#17])
+ *    |  +--child2
+ *    +--LogicalAggregate (groupByExpr=[a#7], outputExpr=[a#7, count(a#7) AS `count(a)`#18]]
+ *      +--child3
+ */
+public class PushCountIntoUnionAll extends OneRewriteRuleFactory {
+    @Override
+    public Rule build() {
+        return logicalAggregate(logicalUnion().when(this::checkUnion))
+                .when(this::checkAgg)
+                .then(this::doPush)
+                .toRule(RuleType.PUSH_COUNT_INTO_UNION_ALL);
+    }
+
+    private Plan doPush(LogicalAggregate<LogicalUnion> agg) {
+        LogicalUnion logicalUnion = agg.child();
+        List<Slot> outputs = logicalUnion.getOutput();
+        Map<Slot, Integer> replaceMap = new HashMap<>();
+        for (int i = 0; i < outputs.size(); i++) {
+            replaceMap.put(outputs.get(i), i);
+        }
+        int childSize = logicalUnion.children().size();
+        List<Expression> upperGroupByExpressions = agg.getGroupByExpressions();
+        List<NamedExpression> upperOutputExpressions = agg.getOutputExpressions();
+        Builder<Plan> newChildren = ImmutableList.builderWithExpectedSize(childSize);
+        Builder<List<SlotReference>> childrenOutputs = ImmutableList.builderWithExpectedSize(childSize);
+        // create the pushed down LogicalAggregate
+        for (int i = 0; i < childSize; i++) {
+            Plan child = logicalUnion.children().get(i);
+            List<Slot> childOutputs = child.getOutput();
+            List<Expression> groupByExpressions = replaceExpressionByUnionAll(upperGroupByExpressions, replaceMap,
+                    childOutputs);
+            List<NamedExpression> outputExpressions = replaceExpressionByUnionAll(upperOutputExpressions, replaceMap,
+                    childOutputs);
+            LogicalAggregate<Plan> logicalAggregate = new LogicalAggregate<>(groupByExpressions, outputExpressions,
+                    child);
+            newChildren.add(logicalAggregate);
+            childrenOutputs.add((List<SlotReference>) (List) logicalAggregate.getOutput());
+        }
+
+        // create the new LogicalUnion
+        LogicalSetOperation newLogicalUnion = logicalUnion.withChildrenAndTheirOutputs(newChildren.build(),
+                childrenOutputs.build());
+        List<NamedExpression> newLogicalUnionOutputs = Lists.newArrayList();
+        for (NamedExpression ce : upperOutputExpressions) {
+            if (ce instanceof Alias) {
+                newLogicalUnionOutputs.add(new SlotReference(ce.getName(), ce.getDataType(), ce.nullable()));
+            } else if (ce instanceof SlotReference) {
+                newLogicalUnionOutputs.add(ce);
+            }
```

Comment:
add else condition to return origin LogicalUnion, means you can not process this cases